### PR TITLE
[RA2 Ch03]: Adding compute resource isolation to Ch03

### LINKED ISSUE #1204: [RA2 Ch02] OS kernel and resource isolation requirements
I think we need to be more specific on the requirements

`req.sec.gen.04 `| General | The Architecture should provide Operating System kernel isolation.
`req.sec.gen.05 `| General | The Architecture must provide compute resources isolation.

Does Operating System kernel isolation here refer to using kata containers, containers in VM, etc?

Does Compute Resource isolation here refer to isolation provided by the standard container runtime?

### PR DESCRIPTION
**Pull request by CsatariGergely**
Partly fixes #1204

### CODE REVIEW

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 1]
```diff
@@ -99,6 +99,10 @@ Where low-level runtimes are focused on the execution of a container within an O
 
 When it comes to Kubernetes, the important interface that we need to consider for container management is the [Kubernetes Container Runtime Interface (CRI)](https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/), which is an interface specification for any container runtime to be able to integrate with the kubelet on a Kubernetes Node. The CRI decouples the kubelet from the runtime that is running in the Host OS, meaning that the code required to integrate kubelet with a container runtime is not part of the kubelet itself (i.e. if a new container runtime comes along, and it uses CRI, it will work with kubelet). Examples of this type of runtime include containerd (with cri plugin) and cri-o, which was built specifically for Kubernetes.
 
+To fullfill `req.sec.gen.04` the architecture should support a container runtime which provides the isoltation of Operating System kernels.
```
Comment (tomkivlin) [inline doc/ref_arch/kubernetes/chapters/chapter03.md]: typo isoltation -> isolation

Comment (rihabbanday) [inline doc/ref_arch/kubernetes/chapters/chapter03.md]: fullfill - > fulfill

**Comment by hdamker** [comment]
> Partly fixes #1204

What does *partly* mean here? Is there another PR planned to fix the rest? (then you shouldn't use "fixes #" as this will close the issue when PR is merged)

@CsatariGergely there are two typos to be fixed as in comment above.

**Comment by walterkozlowski** [comment]
Isn't it really fixing #1204?

**Comment by rabiabdel** [comment]
@CsatariGergely would it be possible please to address comments above?

**Comment by CsatariGergely** [comment]
> > Partly fixes #1204
> 
> What does _partly_ mean here? Is there another PR planned to fix the rest? (then you shouldn't use "fixes #" as this will close the issue when PR is merged)

Yes I plan to create an other pr against Ch04 with the concrete components. I would like to link the pr to the issue, but I do not know how to link the pr-s to the issues without closing them. 

> @CsatariGergely there are two typos to be fixed as in comment above.
Yep, working on it.

File: doc/ref_arch/kubernetes/chapters/chapter03.md (hunk 1 above)
Comment (CsatariGergely) [reply doc/ref_arch/kubernetes/chapters/chapter03.md]: Thanks, fixed

**Pull request merged**